homeassistant/components/sonos/media_player.py

```diff
@@ -176,46 +233,74 @@ def _discovered_player(soco):
     _LOGGER.debug("Adding discovery job")
     hass.async_add_executor_job(_discovery)
 
-    async def async_service_handle(service, data):
+    platform = entity_platform.current_platform.get()
+    management_lock = asyncio.Lock()
+
+    async def async_service_handle(service_call: ServiceCall):
         """Handle dispatched services."""
-        entity_ids = data.get("entity_id")
+        entity_ids = await platform.async_extract_from_service(service_call)
+
+        if not entity_ids:
+            return
+
         entities = hass.data[DATA_SONOS].entities
-        if entity_ids and entity_ids != ENTITY_MATCH_ALL:
-            entities = [e for e in entities if e.entity_id in entity_ids]
-
-        if service == SERVICE_JOIN:
-            master = [
-                e
-                for e in hass.data[DATA_SONOS].entities
-                if e.entity_id == data[ATTR_MASTER]
-            ]
-            if master:
-                await SonosEntity.join_multi(hass, master[0], entities)
-        elif service == SERVICE_UNJOIN:
-            await SonosEntity.unjoin_multi(hass, entities)
-        elif service == SERVICE_SNAPSHOT:
-            await SonosEntity.snapshot_multi(hass, entities, data[ATTR_WITH_GROUP])
-        elif service == SERVICE_RESTORE:
-            await SonosEntity.restore_multi(hass, entities, data[ATTR_WITH_GROUP])
-        else:
-            for entity in entities:
-                if service == SERVICE_SET_TIMER:
-                    call = entity.set_sleep_timer
-                elif service == SERVICE_CLEAR_TIMER:
-                    call = entity.clear_sleep_timer
-                elif service == SERVICE_UPDATE_ALARM:
-                    call = entity.set_alarm
-                elif service == SERVICE_SET_OPTION:
-                    call = entity.set_option
-                elif service == SERVICE_PLAY_QUEUE:
-                    call = entity.play_queue
 
-                hass.async_add_executor_job(call, data)
+        async with management_lock:
```

Comment:
Why is this lock needed?

Reply:
I ported it over from the old code. Can we drop it?

Reply:
Ah, I did not recognize it because it was changed from an `asyncio.Event`.

That `Event` was introduced in #23791 to synchronize the tasks that were added to the dispatcher so that scripts remained sequential.

I think it is not needed here because service calls are already sequential in this case.